util/util.py

```diff
@@ -394,6 +397,90 @@ def parse_bool(text):
     else:
         raise ValueError(f'Cannot parse boolean value from "{text}"')
 
+# from: https://stackoverflow.com/a/5883218
+def get_inheritors(klass):
+    """Get all classes that inherit from klass. """
+    subclasses = set()
+    work = [klass]
+    while work:
+        parent = work.pop()
+        for child in parent.__subclasses__():
+            if child not in subclasses:
+                subclasses.add(child)
+                work.append(child)
+    return subclasses
+
+
+def as_UUID( val, canbenone=True ):
+    """Convert a string or None to a uuid.UUID
+
+    Parameters
+    ----------
+       val : uuid.UUID, str, or None
+         The UUID to be converted.  Will throw a ValueError if val isn't
+         properly formatted.
+
+       canbenone : bool, default True
+         If True, when val is None this function returns None.  If
+         False, when val is None, when val is None this function returns
+         uuid.UUID(''00000000-0000-0000-0000-000000000000').
+
+    Returns
+    -------
+      uuid.UUID or None
+
+    """
+
+    if val is None:
+        if canbenone:
+            return None
+        else:
+            return uuid.UUID( '00000000-0000-0000-0000-000000000000' )
+    if isinstance( val, uuid.UUID ):
+        return val
+    else:
+        return uuid.UUID( val )
+
+
+def as_datetime( string ):
+    """Convert a string to datetime.date with some error checking, allowing a null op.
+
+    Doesn't do anything to take care of timezone aware vs. timezone
+    unaware dates.  It probably should.  Dealing with that is always a
+    nightmare.
```

Comment:
Worthy of an issue on the board?

Reply:
Probably, but in the "nice to have" department.  Right now, if we stick our head in the sand and pretend the earth is flat, we're probably OK.

I *have* been bitten by this in other contexts, before, though.  What I generally like to do is make everything UTC.  If we make all our docker images use UTC, and make the database server use UTC, we're probably fine.  But, it's safer to have timezone aware things and have them all be UTC.

(Oh... yeah, there's a bigger issue, that I think all of our database fields use non-time-zone aware timestamps.  Which, again, is OK if you make sure you're only ever doing UTC.  But, again....)